### PR TITLE
[IBCDPE-219] Create codeql.yml action

### PR DESCRIPTION
**Pull request by thomasyu888**
This PR does code scanning of the GENIE codebase. The reason for this is to reduce code security issues because this is a project that touches phi data. This is part of best practices of application security.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ develop, main ]
```
Comment (thomasyu888) [inline .github/workflows/codeql.yml:16]:
```suggestion
    branches: [ develop, main, ibcdpe-219-codescan ]
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ develop, main, ibcdpe-219-codescan ]
```
Comment (thomasyu888) [inline .github/workflows/codeql.yml]:
```suggestion
    branches: [ develop, main ]
```

**Review by hhunterzinck — APPROVED** [review]

**Comment by thomasyu888** [comment]
Unsure if using LGTM or using codeQL as a github action is a better practice.

**Comment by thomasyu888** [comment]
My initial thought is that LGTM takes way too long and codeQL is really fast - will set this as a standard

**Pull request merged**